pkg/accounts/model/follow.ts

```diff
@@ -0,0 +1,67 @@
+import { Option } from '@mikuroxina/mini-fn';
+
+import type { ID } from '../../id/type.js';
+import type { AccountID } from './account.js';
+
+export interface CreateAccountFollowArgs {
+  fromID: ID<AccountID>;
+  targetID: ID<AccountID>;
+  createdAt: Date;
+  deletedAt?: Date;
+}
+
+/*
+ *
+ * ID: 1  -follow-> ID: 2
+ * => fromID: 1, targetID: 2
+ *
+ * */
+export class AccountFollow {
+  // 必要な情報: フォロー元のアカウントID, フォロー先のアカウントID, created, deleted
+  private constructor(args: CreateAccountFollowArgs) {
+    this.fromID = args.fromID;
+    this.targetID = args.targetID;
+    this.createdAt = args.createdAt;
+    this.deletedAt = !args.deletedAt
+      ? Option.none()
+      : Option.some(args.deletedAt);
+  }
+
+  public static new(args: Omit<CreateAccountFollowArgs, 'deletedAt'>) {
+    return new AccountFollow({ ...args, deletedAt: undefined });
+  }
+
+  public static reconstruct(args: CreateAccountFollowArgs) {
+    return new AccountFollow(args);
+  }
+
+  private readonly fromID: ID<AccountID>;
+
+  get getFromID(): ID<AccountID> {
+    return this.fromID;
+  }
```

Comment:
ゲッター関数に `get` 修飾が付いていますが, これだと `const fromID = follow.getFromID;` のように使うことになるのでおかしい気がします. 通常のメソッドにするか, 名詞的なアクセサにしたほうがよさそうです.
```suggestion
  getFromID(): ID<AccountID> {
    return this.fromID;
  }
```

Reply:
この件ですが, 他のメソッドも同様にお願いします